src/data/zmod/basic.lean

```diff
@@ -281,7 +281,10 @@ begin
   exact dvd_trans h (nat.dvd_sub_mod _),
 end
 
-/-- The canonical ring homomorphism from `zmod n` to a ring of characteristic `n`. -/
+/-- The canonical ring homomorphism from `zmod n` to a ring of characteristic `n`.
+
+See also `zmod.lift` (in `data.zmod.quotient`) for a generalized version working in `add_group`s.
+-/
 def cast_hom (h : m ∣ n) (R : Type*) [ring R] [char_p R m] : zmod n →+* R :=
```

Comment:
It's not clear to me why this requires `ring` in the first place - char_p only requires `add_monoid` and `has_one`.

Reply:
You need at least `add_group` for `cast_add` and it's defined to be a ring hom, so `ring` is indeed required for this specific definition. You are correct that `coe` is a perfectly fine `add_monoid_with_one_hom` even if there is no `mul` in the codomain.

Reply:
Is `cast_add` not true for `add_monoid` or even `add_cancel_monoid`?

Reply:
If `n = 0` then we want to embed `ℤ`, so we need `-1` as well.

I don't actually know if `[add_monoid R] [has_one R]` is enough when `n` is positive - we want that multiples of `1 : R` are cancellative. (`char_p.exists` states that this holds when we are in a `semiring`, and I don't know if the `mul` properties it uses can be avoided.)

Reply:
I think that `[add_group R] [has_one R]` is enough. But if it requires the slightest effort to generalize this, I think it should be parked till a follow-up PR.

Reply:
I tried for a little bit to find an easy way to do this, doesn't seem to be there. So I'll leave it to future work.